src/librustc_trans/trans/closure.rs

```diff
@@ -150,8 +150,9 @@ pub fn get_or_create_closure_declaration<'a, 'tcx>(ccx: &CrateContext<'a, 'tcx>,
         return llfn;
     }
 
-    let path = ccx.tcx().def_path(closure_id);
-    let symbol = mangle_internal_name_by_path_and_seq(path, "closure");
+    let symbol = symbol_names::exported_name(ccx,
+                                             mono_id.def,
+                                             mono_id.params.as_slice());
```

Comment:
Sweet, although I'm not sure whether my PR or this should come first - also, maybe `exported_name` should take an `Instance` (here still called `MonoId`)?